src/Payment/Webhooks/RestApi.php

```diff
@@ -41,33 +41,38 @@ public function registerRoutes()
             [
                 'methods' => 'POST',
                 'callback' => [$this, 'callback'],
-                'permission_callback' => '__return_true',
+                'permission_callback' => [$this, 'verifyMollieSource'],
             ],
         ]);
     }
 
+    public function verifyMollieSource()
+    {
+        $userAgent = sanitize_text_field(wp_unslash($_SERVER['HTTP_USER_AGENT'] ?? ''));
```

Comment:
I think relying on user agent is not a good idea. What is when it will be changed or filtered somehow?

Reply:
Yeah, you are right. This might not be enough, lets discuss on how to approach this better